src/test/java/org/broadinstitute/hellbender/utils/baq/BAQUnitTest.java

```diff
@@ -1,26 +1,234 @@
 package org.broadinstitute.hellbender.utils.baq;
 
+import com.google.common.base.Strings;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.SAMUtils;
 import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.engine.ReferenceDataSource;
+import org.broadinstitute.hellbender.engine.ReferenceFileSource;
+import org.broadinstitute.hellbender.engine.ReferenceMemorySource;
 import org.broadinstitute.hellbender.utils.SimpleInterval;
 import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
 import org.broadinstitute.hellbender.utils.test.BaseTest;
 import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
-public final class BAQUnitTest extends BaseTest{
+import java.io.File;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class BAQUnitTest extends BaseTest {
+
+    private SAMFileHeader header;
+    private final int startChr = 1;
+    private final int numChr = 2;
+    private final int chrSize = 1000;
+
+    @BeforeMethod
+    public void before() {
+        header = ArtificialReadUtils.createArtificialSamHeader(numChr, startChr, chrSize);
```

Comment:
Why even have a shared header? Just have this method return the header, rename to `createHeader()` and remove `@BeforeMethod`, and have the tests call it. Shared mutable data in test classes is a source of bugs, and limits our options for test suite parallelism.